onadata/apps/logger/migrations/0030_backfill_lost_monthly_counters.py

```diff
@@ -15,6 +15,17 @@ def populate_missing_monthly_counters(apps, schema_editor):
     if not DailyXFormSubmissionCounter.objects.all().exists():
         return
 
+    # Associate each daily counter with user=None with a user based on its xform
+    for counter in DailyXFormSubmissionCounter.objects.filter(user=None).iterator:
+        if counter.xform and counter.xform.user:
+            has_duplicate = DailyXFormSubmissionCounter.objects.filter(
+                date=counter.date, xform=counter.xform
+            ).exclude(user=None).exists()
+            # don't add a user to duplicate counters, so they get deleted in the next step
+            if not has_duplicate:
+                counter.user = counter.xform.user
+                counter.save()
+
```

Comment:
I would prefer to use `bulk_update()` to be more efficient. 

Something like that: 
```python 
    batch = []
    batch_size = 5000 
    for counter in (
        DailyXFormSubmissionCounter.objects.filter(user=None)
        .exclude(xform=None)
        .iterator(chunk_size=batch_size)
    ):
        counter.user = counter.xform.user
        if DailyXFormSubmissionCounter.objects.filter(
            date=counter.date, xform=counter.xform
        ).exclude(user=None).exists():
            continue
        batch.append(counter)
        if len(batch) >= batch_size:
            DailyXFormSubmissionCounter.objects.bulk_update(batch, ['user_id'])
            batch = []
    if batch:
        DailyXFormSubmissionCounter.objects.bulk_update(batch, ['user_id'])
```
What do you think?